src/DataTypes/Serializations/ISerialization.cpp

```diff
@@ -165,6 +172,11 @@ String getNameForSubstreamPath(
 {
     using Substream = ISerialization::Substream;
 
+    auto context = CurrentThread::getQueryContext();
+    if (!context)
+        context = Context::getGlobalContextInstance();
+    bool escape_variant = context->getSettingsRef()[Setting::save_variant_with_escaped_filename];
```

Comment:
I think we can just use global context and don't even check for query context. I don't see a good scenario when this setting can be changed in the query